server/vault.api.js

```diff
@@ -0,0 +1,128 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const HomebrewModel = require('./homebrew.model.js').model;
+
+const router = express.Router();
+
+const buildTitleConditions = (title) => {
+    if (!title) return {};
+    return {
+        $text: {
+            $search: title,
+            $caseSensitive: false,
+        },
+    };
+};
+
+const buildAuthorConditions = (author) => {
+    if (!author) return {};
+    return { authors: author };
+};
+
+//"$and": [ {"published": true}, {"$text": { "$search": "titleString", "$caseSensitive": false } }, { "authors" : "authorString"}]
+//is a good example of a query constructed with this function
+
+const handleErrorResponse = (res, error, functionName) => {
+    const status = error.response?.status || 500;
+    const message =
+        status === 503 ? 'Service Unavailable' : 'Internal Server Error';
+
+    console.error(`Error in ${functionName}:`, error);
+
+    return res.status(status).json({
+        errorCode: status.toString(),
+        message: `Error in function ${functionName}: ${message}`,
+    });
+};
```

Comment:
Instead of a whole separate error handler, can we just utilize the central error handler we already have in `app.js`? That's what `asyncHandler()` is for. It passes error objects along. All you need to do is throw the error object and it will land there and be sent. You can customize any of the properties in the object. The whole error object will be logged, it already has logic to handle the response status, and if you add an `HBErrorCode` we can track where it came from.

See:

https://github.com/naturalcrit/homebrewery/blob/c41116e6c8eadc418b1dcf799dd3a83c74d5bcae/server/homebrew.api.js#L136-L138

Reply:
replaced with trows